### PR TITLE
Adds unit test for imu preintegration of a single step

### PR DESCRIPTION
**Pull request by raabuchanan**
Related to issue #627 

This adds a unit test which integrates simulated data from an IMU in a foot while to foot takes a single step. See issue for related dataset.

Currently the unit test actually passes which is in contradiction to what I am seeing in the issue. I need to investigate the problem further but wanted to put up this unit test to keep everyone in the loop. Also someone may spot something wrong that I'm doing. I will write more unit tests to try and understand the problem.

### CODE REVIEW

**Comment by varunagrawal** [comment]
I believe we can merge all 3 files into 1 so we have a comprehensive test file that is easy to modify. While I like the structure a lot, it seems like a lot of added complexity.

**Comment by varunagrawal** [comment]
@raabuchanan I've made some updates to the code to make it agnostic to users. I'd love to work with you on this PR and figure this issue out.

Currently, I am a bit confused on what are we testing for specifically? Is there a specific assertion you want to test for?

**Comment by dellaert** [comment]
@varunagrawal , please help land? Also, don’t force-push :-)

**Comment by varunagrawal** [comment]
Since @raabuchanan got the Imu impact working, what should we be testing for? The same thing as before?

@raabuchanan please could you refresh my memory on the goal of this test? I believe you had a slide on it.

**Comment by raabuchanan** [comment]
Hey @varunagrawal yes I fixed the problem on my side. I've just pushed a small cleanup now.

**Comment by raabuchanan** [comment]
Hi @varunagrawal yes I fixed the issue on my side and this passes now. This test checks a sequence of simulated IMU data that the relative Z position is zero. It also checks that there are no rotations. I will push a small cleanup soon but I can add more documentation if you want.

**Comment by varunagrawal** [comment]
Awesome. I'll clean it up a little bit and then we can merge.

File: tests/testImuPreintegration.cpp [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+      results.push_back(atof(str.c_str()));
+    }
+    ImuMeasurement measurement;
+    measurement.dt = static_cast<size_t>(1e9 * (1 / rate));
+    measurement.time = results[2];
+    measurement.I_a_WI = {results[29], results[30], results[31]};
+    measurement.I_w_WI = {results[17], results[18], results[19]};
+    imuMeasurements.push_back(measurement);
+
+    // cout << "IMU measurement " << measurement << endl;
```
Comment (varunagrawal) [inline tests/testImuPreintegration.cpp:68]: We should get rid of the print statements. Tests should be silent and fast.

File: tests/testImuPreintegration.cpp [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  for (size_t n = 1; n < imuMeasurements.size(); n++) {
+    // integrate
+    imuPreintegrated.integrateMeasurement(imuMeasurements[n].I_a_WI,
+                                          imuMeasurements[n].I_w_WI, 1 / rate);
+    // predict
+    propState = imuPreintegrated.predict(initialNavState, priorImuBias);
+    position = propState.pose().translation();
+    velocity = propState.velocity();
+    // cout << "IMU Position " << position.transpose() << endl;
+    // cout << "IMU Velocity " << velocity.transpose() << endl;
```
Comment (varunagrawal) [inline tests/testImuPreintegration.cpp:111]: Need to remove these prints.

**Review by varunagrawal — APPROVED** [review]
Very minor comments, but LGTM.

**Pull request merged**